src/d3d12/nvapi_d3d12_device.cpp

```diff
@@ -105,17 +148,45 @@ namespace dxvk {
         return deviceExt;
     }
 
-    Com<ID3D12GraphicsCommandListExt> NvapiD3d12Device::GetCommandListExt(ID3D12GraphicsCommandList* commandList) {
+    d3d12_graphics_command_list_ext_iface* NvapiD3d12Device::GetCommandListExt(ID3D12GraphicsCommandList* commandList, uint32_t* interfaceVersion) {
         std::scoped_lock lock(m_CommandListMutex);
         auto it = m_CommandListMap.find(commandList);
-        if (it != m_CommandListMap.end())
-            return it->second;
-
-        Com<ID3D12GraphicsCommandListExt> commandListExt;
-        if (FAILED(commandList->QueryInterface(IID_PPV_ARGS(&commandListExt))))
+        if (it != m_CommandListMap.end()) {
+            *interfaceVersion = it->second.second;
+            return it->second.first;
+        }
+
+        d3d12_graphics_command_list_ext_iface* iface = nullptr;
+
+        // Grab the highest interface possible
+        do {
+            ID3D12GraphicsCommandListExt1* commandListExt1 = nullptr;
+            commandList->QueryInterface(IID_PPV_ARGS(&commandListExt1));
+
+            if (commandListExt1) {
```

Comment:
By convention:

```suggestion
            if (SUCCEEDED(commandList->QueryInterface(IID_PPV_ARGS(&commandListExt1)))) {
```

Same below.